diff --git a/panopoly_images.make b/panopoly_images.make
index a9d572d..ba543ef 100644
--- a/panopoly_images.make
+++ b/panopoly_images.make
@@ -11,7 +11,7 @@ projects[manualcrop][download][type] = git
 projects[manualcrop][download][revision] = d6c449d
 projects[manualcrop][download][branch] = 7.x-1.x
 projects[manualcrop][patch][2521962] = https://www.drupal.org/files/issues/manualcrop_dont_show_crop_for_video-2521962-1.patch
-
+projects[manualcrop][patch][2560427] = https://www.drupal.org/files/issues/manualcrop-alt-element-may-have-no-weight-2560427-1.patch
 
 ; jquery.imagesLoaded library for manualcrop
 libraries[jquery.imagesloaded][download][type] = file

