diff --git a/panopoly_core.make b/panopoly_core.make
index 63cf5ba..7fa1007 100644
--- a/panopoly_core.make
+++ b/panopoly_core.make
@@ -62,9 +62,8 @@ projects[views_bulk_operations][subdir] = contrib
 
 ; The Usual Suspects
 
-projects[pathauto][version] = 1.2
+projects[pathauto][version] = 1.3
 projects[pathauto][subdir] = contrib
-projects[pathauto][patch][936222] = http://drupal.org/files/issues/pathauto-persist-936222-195-pathauto-state.patch
 
 projects[token][version] = 1.6
 projects[token][subdir] = contrib
@@ -86,7 +85,7 @@ projects[transliteration][subdir] = contrib
 
 ; Field modules
 
-projects[date][version] = 2.9-rc1
+projects[date][version] = 2.9
 projects[date][subdir] = contrib
 
 projects[entityreference][version] = 1.x-dev
@@ -117,13 +116,10 @@ projects[features][subdir] = contrib
 projects[strongarm][version] = 2.0
 projects[strongarm][subdir] = contrib
 
-projects[defaultconfig][version] = 1.0-alpha9
+projects[defaultconfig][version] = 1.0-alpha11
 projects[defaultconfig][subdir] = contrib
-projects[defaultconfig][patch][2042799] = http://drupal.org/files/default_config_delete_only_if_overriden.patch
-projects[defaultconfig][patch][2043307] = http://drupal.org/files/defaultconfig_include_features_file.patch
-projects[defaultconfig][patch][2008178] = http://drupal.org/files/defaultconfig-rebuild-filters-2008178-4_0.patch
-projects[defaultconfig][patch][1861054] = http://drupal.org/files/fix-defaultconfig_rebuild_all.patch
-projects[defaultconfig][patch][1900574] = http://drupal.org/files/1900574.defaultconfig.undefinedindex_11.patch
+projects[defaultconfig][patch][2008178] = https://www.drupal.org/files/issues/defaultconfig-rebuild-2008178-14.patch
+projects[defaultconfig][patch][1900574] = https://www.drupal.org/files/issues/1900574.defaultconfig.undefinedindex_13.patch
 
 projects[defaultcontent][version] = 1.0-alpha9
 projects[defaultcontent][subdir] = contrib
@@ -154,6 +150,5 @@ projects[distro_update][subdir] = contrib
 projects[features_override][version] = 2.0-rc3
 projects[features_override][subdir] = contrib
 
-projects[uuid][version] = 1.0-alpha6
+projects[uuid][version] = 1.0-beta1
 projects[uuid][subdir] = contrib
-projects[uuid][patch][2356305] = https://www.drupal.org/files/issues/2356305-5.patch
