diff --git a/panopoly_wysiwyg.make b/panopoly_wysiwyg.make
index 5ebcba2..41f2459 100644
--- a/panopoly_wysiwyg.make
+++ b/panopoly_wysiwyg.make
@@ -13,18 +13,16 @@ projects[wysiwyg][patch][1802394] = http://drupal.org/files/wysiwyg-1802394-4.pa
 projects[wysiwyg][patch][934976] = https://drupal.org/files/wysiwyg-one-format.934976.23-alt.patch
 projects[wysiwyg][patch][1757684] = https://www.drupal.org/files/issues/wysiwyg-1757684.28.patch
 
-projects[wysiwyg_filter][version] = 1.6-rc2
+projects[wysiwyg_filter][version] = 1.6-rc3
 projects[wysiwyg_filter][subdir] = contrib
-projects[wysiwyg_filter][patch][1687794] = https://www.drupal.org/files/wysiwyg_filter-1687794-1-skip-validation-if-filter-disabled.patch
 
 ; The WYSIWYG Helpers
 
-projects[linkit][version] = 3.4
+projects[linkit][version] = 3.5
 projects[linkit][subdir] = contrib
 
-projects[image_resize_filter][version] = 1.15
+projects[image_resize_filter][version] = 1.16
 projects[image_resize_filter][subdir] = contrib
-projects[image_resize_filter[patch][1929710] = https://www.drupal.org/files/issues/image_resize_filter-query_string-1929710-14-D7.patch
 
 projects[caption_filter][version] = 1.x-dev
 projects[caption_filter][subdir] = contrib

