diff --git a/panopoly_wysiwyg.make b/panopoly_wysiwyg.make
index 5ebcba2..06d2f1d 100644
--- a/panopoly_wysiwyg.make
+++ b/panopoly_wysiwyg.make
@@ -22,9 +22,8 @@ projects[wysiwyg_filter][patch][1687794] = https://www.drupal.org/files/wysiwyg_
 projects[linkit][version] = 3.4
 projects[linkit][subdir] = contrib

-projects[image_resize_filter][version] = 1.15
+projects[image_resize_filter][version] = 1.16
 projects[image_resize_filter][subdir] = contrib
-projects[image_resize_filter[patch][1929710] = https://www.drupal.org/files/issues/image_resize_filter-query_string-1929710-14-D7.patch

 projects[caption_filter][version] = 1.x-dev
 projects[caption_filter][subdir] = contrib
