diff --git a/panopoly_widgets.make b/panopoly_widgets.make
index b3602bd..db0dbc1 100644
--- a/panopoly_widgets.make
+++ b/panopoly_widgets.make
@@ -18,15 +18,19 @@ projects[menu_block][subdir] = contrib
 
 ; Panopoly - Contrib - Files & Media
 
-projects[file_entity][version] = 2.0-beta2
+projects[file_entity][version] = 2.0-beta3
 projects[file_entity][subdir] = contrib
+projects[file_entity][patch][2360043] = https://www.drupal.org/files/issues/file_entity-destination-token-decode-2360043-1.patch
+projects[file_entity][patch][2386015] = https://www.drupal.org/files/issues/2386015-file_entity-trigger-1.patch
 
-projects[media][version] = 2.0-beta1
+projects[media][version] = 2.0-beta2
 projects[media][subdir] = contrib
 projects[media][patch][2126697] = https://www.drupal.org/files/issues/media_wysiwyg_2126697-53.patch
 projects[media][patch][2308487] = https://www.drupal.org/files/issues/media-alt-title-double-encoded-2308487-2.patch
-projects[media][patch][2084287] = http://www.drupal.org/files/issues/media-file-name-focus-2084287-2.patch
 projects[media][patch][2534724] = https://www.drupal.org/files/issues/media-browser_opens_twice-2534724-53.patch
+projects[media][patch][951004] = https://www.drupal.org/files/issues/media-multiselect-cardinality-951004-219-D7.patch
+projects[media][patch][2473481] = http://drupal.org/files/issues/clicking_images_in-2473481-1.patch
+projects[media][patch][2528428] = https://www.drupal.org/files/issues/media_browser_view_javascript_not_loaded-2528428-1.patch
 
 projects[media_youtube][version] = 3.0
 projects[media_youtube][subdir] = contrib

