diff --git a/panopoly_core.make b/panopoly_core.make
index 08fe91e..9d25c52 100644
--- a/panopoly_core.make
+++ b/panopoly_core.make
@@ -15,12 +15,14 @@ projects[ctools][patch][2448989] = https://www.drupal.org/files/issues/2448989-c
 projects[ctools][patch][2483415] = https://www.drupal.org/files/issues/ctools-more_than_one_comment_pager-2483415-1.patch
 projects[ctools][patch][2493041] = https://www.drupal.org/files/issues/ctools-n2493041-3.patch
 projects[ctools][patch][2555469] = https://www.drupal.org/files/issues/2555469-ctools-comments-node_tag_new-2.patch
+projects[ctools][patch][2449245] = https://www.drupal.org/files/issues/ctools-cloning-page-manager-duplicates-uuids-2449245-07.patch

 projects[panels][version] = 3.5
 projects[panels][subdir] = contrib
 projects[panels][patch][2448825] = https://www.drupal.org/files/issues/panels-export-indentation-2448825-1.patch
 projects[panels][patch][2390803] = https://www.drupal.org/files/issues/panels-focus-add-content-tab-2390803-13.patch
 projects[panels][patch][1588212] = https://www.drupal.org/files/issues/panels-1588212-10.patch
+projects[panels][patch][2253919] = https://www.drupal.org/files/issues/panels-re-generated-uuids-on-clone.patch

 projects[panels_breadcrumbs][version] = 2.2
 projects[panels_breadcrumbs][subdir] = contrib
