diff --git a/panopoly_core.make b/panopoly_core.make
index bbf9e54..d4dfb4e 100644
--- a/panopoly_core.make
+++ b/panopoly_core.make
@@ -67,9 +67,8 @@ projects[views_bulk_operations][subdir] = contrib

 ; The Usual Suspects

-projects[pathauto][version] = 1.2
+projects[pathauto][version] = 1.3
 projects[pathauto][subdir] = contrib
-projects[pathauto][patch][936222] = http://drupal.org/files/issues/pathauto-persist-936222-195-pathauto-state.patch

 projects[token][version] = 1.6
 projects[token][subdir] = contrib
@@ -128,6 +127,8 @@ projects[strongarm][subdir] = contrib
 projects[defaultconfig][version] = 1.0-alpha11
 projects[defaultconfig][subdir] = contrib
 projects[defaultconfig][patch][1900574] = http://drupal.org/files/1900574.defaultconfig.undefinedindex_11.patch
+projects[defaultconfig][patch][2008178] = https://www.drupal.org/files/issues/defaultconfig-rebuild-2008178-14.patch
+projects[defaultconfig][patch][1900574] = https://www.drupal.org/files/issues/1900574.defaultconfig.undefinedindex_13.patch

 projects[defaultcontent][version] = 1.0-alpha9
 projects[defaultcontent][subdir] = contrib
@@ -158,6 +159,5 @@ projects[distro_update][subdir] = contrib
 projects[features_override][version] = 2.0-rc3
 projects[features_override][subdir] = contrib

-projects[uuid][version] = 1.0-alpha6
+projects[uuid][version] = 1.0-beta1
 projects[uuid][subdir] = contrib
-projects[uuid][patch][2356305] = https://www.drupal.org/files/issues/2356305-5.patch
